scaleway/resource_secret_version.go

```diff
@@ -0,0 +1,204 @@
+package scaleway
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	secret "github.com/scaleway/scaleway-sdk-go/api/secret/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewaySecretVersion() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewaySecretVersionCreate,
+		ReadContext:   resourceScalewaySecretVersionRead,
+		UpdateContext: resourceScalewaySecretVersionUpdate,
+		DeleteContext: resourceScalewaySecretVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultSecretTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"secret_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The secret ID associated with this version",
+				DiffSuppressFunc: diffSuppressFuncLocality,
+			},
+			"data": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "The data payload of your secret version",
+				Sensitive:        true,
+				ForceNew:         true,
+				DiffSuppressFunc: diffSuppressFuncBase64,
+				StateFunc: func(i interface{}) string {
+					return base64Encoded([]byte(i.(string)))
+				},
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the secret version",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the secret version",
+			},
+			"revision": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The revision of secret version",
+			},
+			"with_access": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Enable access to the payload secret",
+				Default:     false,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date and time of secret version's creation (RFC 3339 format)",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date and time of secret version's creation (RFC 3339 format)",
+			},
+			"region": regionSchema(),
+		},
+	}
+}
+
+func resourceScalewaySecretVersionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := secretAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secretID := expandID(d.Get("secret_id").(string))
+	payloadSecret := []byte(d.Get("data").(string))
+	if isBase64Encoded(payloadSecret) {
```

Comment:
proposal: return an error when it's not base64 and only manipulate base64 for this attribute